sdk/tools/azure-openrewrite/src/main/resources/META-INF/rewrite/rewrite.yml

```diff
@@ -0,0 +1,255 @@
+  # Before:
+  # import com.azure.core.credential.*;
+  # After:
+  # import io.clientcore.core.credential.*;
+  - org.openrewrite.java.ChangePackage:
+      oldPackageName: com.azure.core.credential
+      newPackageName: io.clientcore.core.credential
+
+  # Recipes that update all instances of azure-core exceptions where
+  # use is equivalent and type/name change is sufficient/safe.
+  # Update to azure-core-v2 version
+  # Before:
+  # import com.azure.core.exception.ClientAuthenticationException;
+  # After:
+  # import com.azure.core.v2.exception.ClientAuthenticationException;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.exception.ClientAuthenticationException
+      newFullyQualifiedTypeName: com.azure.core.v2.exception.ClientAuthenticationException
+  # Before:
+  # import com.azure.core.exception.ResourceModifiedException;
+  # After:
+  # import com.azure.core.v2.exception.ResourceModifiedException;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.exception.ResourceModifiedException
+      newFullyQualifiedTypeName: com.azure.core.v2.exception.ResourceModifiedException
+  # Before:
+  # import com.azure.core.exception.ResourceNotFoundException;
+  # After:
+  # import com.azure.core.v2.exception.ResourceNotFoundException;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.exception.ResourceNotFoundException
+      newFullyQualifiedTypeName: com.azure.core.v2.exception.ResourceNotFoundException
+  # Update to clientcore version
+  # Before:
+  # import com.azure.core.exception.HttpResponseException;
+  # After:
+  # import io.clientcore.core.http.exception.HttpResponseException;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.exception.HttpResponseException
+      newFullyQualifiedTypeName: io.clientcore.core.http.exception.HttpResponseException
+  # End azure-core-exception change type Recipes
+
+  # Recipe that changes all instances of com.azure.core.util.Configuration
+  # to io.clientcore.core.util.configuration.Configuration
+  # Before:
+  # import com.azure.core.util.Configuration;
+  # After:
+  # import io.clientcore.core.util.configuration.Configuration;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.util.Configuration
+      newFullyQualifiedTypeName: io.clientcore.core.util.configuration.Configuration
+
+  # azure-core v2 BinaryData.toObject throws an IOException. This recipe encloses all calls in
+  # a try-catch block.
+  # Before:
+  # List<TranslatedTextItem> result = binaryDataResponse.getValue().toObject( /* Any Args */);
+  # After:
+  # List<TranslatedTextItem> result = null;
+  # try {
+  #     result = binaryDataResponse.getValue().toObject( /* Any Args */);
+  # } catch (IOException e) {
+  #     e.printStackTrace();
+  # }
+  # WARNING: This recipe MUST be run before com.azure.openrewrite.TypeReferenceRecipe.
```

Comment:
This concerns me that someone using this must read documentation like this to know how to migrate. Is there any way to remove this requirement where this must run before `TypeReferenceRecipe`?

Reply:
@alzimmermsft The end-user would not see this. This is a warning for us as the maintainers since recipes execute sequentially.